common/persistence/cassandra/cassandraMetadataPersistence.go

```diff
@@ -100,8 +100,16 @@ type (
 	}
 )
 
-// newMetadataPersistence is used to create an instance of HistoryManager implementation
-func newMetadataPersistence(cfg config.Cassandra, clusterName string, logger log.Logger) (p.MetadataStore,
+// NewMetadataPersistenceWithSession is used to create an instance of HistoryManager implementation
+func NewMetadataPersistenceWithSession(session *gocql.Session, clusterName string, logger log.Logger) p.MetadataStore {
+	return &cassandraMetadataPersistence{
+		cassandraStore:     cassandraStore{session: session, logger: logger},
+		currentClusterName: clusterName,
+	}
+}
+
+// NewMetadataPersistence is used to create an instance of HistoryManager implementation
+func NewMetadataPersistence(cfg config.Cassandra, clusterName string, logger log.Logger) (p.MetadataStore,
```

Comment:
nit: I don't think we need to expose this as public method. I would keep it private until we need it.